### PR TITLE
Add dev container

### PR DESCRIPTION
**Pull request by thecode**
Add dev container to allow using `script/setup` on non Linux based environments (e.g., Windows)

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.177.0/containers/python-3
+{
+	"name": "Python 3",
+	"build": {
+		"dockerfile": "Dockerfile",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:6]: No need for a dockerfile, set `"image": "mcr.microsoft.com/vscode/devcontainers/python:0-3.10"` instead

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:1]: Use `.devcontainer.json` in the root instead of in this subdir

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.177.0/containers/python-3
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:2]: Remove comments

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{
+	"name": "Python 3",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:4]:
```suggestion
	"name": "home-assistant/intents",
```

File: .devcontainer/devcontainer.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		"context": "..",
+		"args": {
+			// Update 'VARIANT' to pick a Python version: 3, 3.6, 3.7, 3.8, 3.9
+			"VARIANT": "3.10",
+			// Options
+			"INSTALL_NODE": "false",
+			"NODE_VERSION": "lts/*"
+		}
+	},
+	// Set *default* container specific settings.json values on container create.
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:16]: Remove comments

File: .devcontainer/devcontainer.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
+		"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
+		"python.linting.banditPath": "/usr/local/py-utils/bin/bandit",
+		"python.linting.flake8Path": "/usr/local/py-utils/bin/flake8",
+		"python.linting.mypyPath": "/usr/local/py-utils/bin/mypy",
+		"python.linting.pycodestylePath": "/usr/local/py-utils/bin/pycodestyle",
+		"python.linting.pydocstylePath": "/usr/local/py-utils/bin/pydocstyle",
+		"python.linting.pylintPath": "/usr/local/py-utils/bin/pylint"
+	},
+	// Add the IDs of extensions you want installed when the container is created.
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:33]: Remove comments

File: .devcontainer/devcontainer.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+	"extensions": [
+		"ms-python.python",
+		"ms-python.vscode-pylance"
+	],
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:38]: Remove comments

File: script/setup [hunk 8]
```diff
@@ -31,7 +31,7 @@ pushd "${base_dir}/parser" 2>/dev/null
 pip3 install -e '.'
 popd 2>/dev/null
 
-pip3 install -r "${base_dir}/requirements.txt"
+pip3 install -r "${base_dir}/requirements_dev.txt"
```
Comment (ludeeus) [inline script/setup:34]: Why split the requirements file, when you are just referencing the combined again here?

File: .devcontainer/devcontainer.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+	// "forwardPorts": [],
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "script/setup",
+	// Comment out connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:42]: Remove comments

File: .devcontainer/devcontainer.json [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+		"python.linting.enabled": true,
+		"python.linting.pylintEnabled": true,
+		"python.formatting.autopep8Path": "/usr/local/py-utils/bin/autopep8",
+		"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
+		"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
+		"python.linting.banditPath": "/usr/local/py-utils/bin/bandit",
+		"python.linting.flake8Path": "/usr/local/py-utils/bin/flake8",
+		"python.linting.mypyPath": "/usr/local/py-utils/bin/mypy",
+		"python.linting.pycodestylePath": "/usr/local/py-utils/bin/pycodestyle",
+		"python.linting.pydocstylePath": "/usr/local/py-utils/bin/pydocstyle",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:30]: pydocstyle is not installed

File: .devcontainer/devcontainer.json [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+		"python.languageServer": "Pylance",
+		"python.linting.enabled": true,
+		"python.linting.pylintEnabled": true,
+		"python.formatting.autopep8Path": "/usr/local/py-utils/bin/autopep8",
+		"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
+		"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
+		"python.linting.banditPath": "/usr/local/py-utils/bin/bandit",
+		"python.linting.flake8Path": "/usr/local/py-utils/bin/flake8",
+		"python.linting.mypyPath": "/usr/local/py-utils/bin/mypy",
+		"python.linting.pycodestylePath": "/usr/local/py-utils/bin/pycodestyle",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:29]: pycodestyle is not installed

File: .devcontainer/devcontainer.json [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash",
+		"python.pythonPath": "/usr/local/bin/python",
+		"python.languageServer": "Pylance",
+		"python.linting.enabled": true,
+		"python.linting.pylintEnabled": true,
+		"python.formatting.autopep8Path": "/usr/local/py-utils/bin/autopep8",
+		"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
+		"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
+		"python.linting.banditPath": "/usr/local/py-utils/bin/bandit",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:26]: bandit is not installed

File: .devcontainer/devcontainer.json [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+		}
+	},
+	// Set *default* container specific settings.json values on container create.
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash",
+		"python.pythonPath": "/usr/local/bin/python",
+		"python.languageServer": "Pylance",
+		"python.linting.enabled": true,
+		"python.linting.pylintEnabled": true,
+		"python.formatting.autopep8Path": "/usr/local/py-utils/bin/autopep8",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:23]: autopep8 is not installed

File: .devcontainer/devcontainer.json [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+		"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
+		"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
```
Comment (ludeeus) [inline .devcontainer/devcontainer.json:25]: yapf is not installed

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: script/setup (hunk 8 above)
Comment (thecode) [reply script/setup:34]: This allow simpler management of linters and doesn't force lib users to install linters, I don't mind combining it back if you think it is better

Comment (ludeeus) [reply script/setup:34]: Lib users? This is not a lib?
We should eventually (with pre-commit) force linting anyway.

Comment (thecode) [reply script/setup:34]: Ok, I will combine back, pre-commit was my next planned PR anyhow

**Review by ludeeus — APPROVED** [review]

**Comment by thecode** [comment]
Thanks!

**Pull request merged**